docs/DeveloperGuide.md

```diff
@@ -441,43 +441,107 @@ considering code quality and end user experience.
+
+Given below is an example usage scenario, together with a sequence diagram, to show how the assign macro mechanism behaves at each step.
+![Assign Macro Command Sequence Diagram](images/AssignMacroCommandSequenceDiagram.png)
+
+Step 1. The user discovers that he frequently updates a specific warehouse and decides to create a new macro with the alias "uwm" for the command string "update ct/w n/MainWarehouse" so that he can enter a shorter command in the future.
+The user does this by executing the `assignmacro a/uwm cs/update ct/w n/MainWarehouse` command. The input string will then be passed to the `AssignMacroCommandParser`.
+
+Step 2. By matching the prefixes provided, `AssignMacroCommandParser#parse` then attempts to create a new instances of `Alias` and `SavedCommandString` after matching the prefixes, and throws
+an exception to be displayed on the GUI if the alias or command string supplied by the user is invalid, or if any of them is not supplied at all. If all prefixes are parsed without error,
+a `Macro` is created from the `Alias` and `SavedCommandString` instances. Then, a new `AssignMacroCommand` instance is created with the new `Macro`.
+
+Step 3. The `AssignMacroCommand#execute` method will then be called with the `model` instance. The method will first check if there is any existing macro in the model that uses the same alias,
+throwing an exception which will be shown on the GUI as an error message if that is true. Otherwise, the new macro will be added to the model.
+
+Step 4. The `AssignMacroCommand#execute` then passes a `CommandResult` with a success message back to the `LogicManager`. Finally, the model is saved and the GUI is updated with the success message.
+The user now updates the quantity of the product "Panadol" in the aforementioned warehouse by simply executing the command `uwm pd/Panadol`.
+
+The following activity diagram summarizes what happens when a user assigns a macro:
+![Assign Macro Command Activity Diagram](images/AssignMacroCommandActivityDiagram.png)
+
+### Remove Macro feature
+
+#### What the remove macro feature does
+
+The remove macro feature allows users to be able to remove an existing macro with a specific alias.
+
+#### How it is implemented
+
+The remove macro mechanism is facilitated by 2 components: `RemoveMacroCommandParser` and `RemoveMacroCommand`.
+`RemoveMacroCommandParser`'s job is to parse the user input to generate the correct objects for the `RemoveMacroCommand`.
+`RemoveMacroCommand`'s job is to execute the main logic for updating the model with the specified macro removed.
+
+Given below is an example usage scenario, together with a sequence diagram, to show how the remove macro mechanism behaves at each step.
+![Remove Macro Command Sequence Diagram](images/RemoveMacroCommandSequenceDiagram.png)
+
+Step 1. The user decides that he no longer needs the macro with the alias "uwm" and decides to remove it. He does this by executing the `removemacro uwm` command.
```

Comment:
Would he/she be better here instead of he? 🤔

Reply:
If the above were to be adopted,

If the user no longer needs the macro with the alias "uwm" and decides to remove it, it can be done by executing the `removemacro uwm` command.